xc7/utils/prjxray_db_cache.py

```diff
@@ -0,0 +1,53 @@
+"""
+This is a database cache. Its a kind of proxy object for Python sqlite3 connection
+which operates on a memory copy of the database.
+
+Upon object creation the database is "backed up" to memory. All subsequent
+operations are then pefromed on this copy which yields in performance increase.
+
+It is important to explicitly call the close() method once database connection
+is no longer needed. The method "backs up" the database back to disk.
+"""
+import sqlite3
+
+# =============================================================================
+
+class DatabaseCache(object):
+
+    def __init__(self, file_name):
```

Comment:
Depending on the results, let's make the cache optional in a transparent way.  Doing this will likely increase the memory usage significantly, but I'd like to get profile results.

Reply:
Given that peak memory usage didn't increase when using the in-memory cache, I think we can skip making it optional.